cozy/codegen/cxx.py

```diff
@@ -274,6 +274,8 @@ def construct_concrete(self, t : Type, e : Exp, out : Exp):
             return SEscape("{indent}{lhs} = {rhs};\n", ["lhs", "rhs"], [out, e])
         elif is_scalar(t):
             return SEscape("{indent}{lhs} = {rhs};\n", ["lhs", "rhs"], [out, e])
+        elif isinstance(t, TTuple):
+            return SEscape("{indent}{lhs} = {rhs};\n", ["lhs", "rhs"], [out, e])
```

Comment:
@Calvin-L I believe this also fixes the problem in #70, thus I am a bit confused here

Reply:
See my note in #72 about aliases. I think this solution is dangerous and you really need to create a `new` tuple object.

Reply:
> you really need to create a new tuple object.

How exactly? `rhs` will be visited and turned into `new __typeXXX`. However, this is not apparent from the code

Reply:
I believe this has now been correctly addressed in #73. 😄